File: Mastering_Swift4/13Using Mix and Match/ObjectiveCExample/ObjectiveCExample/ViewController.m
#import "ViewController.h"
#import "ObjectiveCExample-Swift.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

-(IBAction)messagePressed:(id)sender {
    MessageBuilder *mb = [[MessageBuilder alloc] init];

    self.messageView.text = [mb getPersonalizedMessageWithName:self.nameField.text];
}


@end

File: Beginning iPhone Development with Swift/4ControlFun/4ControlFun/ViewController.swift
//
//  ViewController.swift
//  4ControlFun
//
//  Created by 万畅 on 15/8/24.
//  Copyright © 2015年 abelwan. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    @IBOutlet weak var nameField: UITextField!

    @IBOutlet weak var numberField: UITextField!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func textFieldDoneEditing(sender: UITextField) {
        sender.resignFirstResponder()
    }
    
    @IBAction func backgroundTap(sender:UIControl){
     nameField.resignFirstResponder()
     numberField.resignFirstResponder()
    }
    

}
